File: src/org/elixir_lang/leex/file/Type.java
package org.elixir_lang.leex.file;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;

public class Type extends org.elixir_lang.eex.file.Type {
    public static final LanguageFileType INSTANCE = new Type();

    @NotNull
    @Override
    public String getName() {
        return "Live Embedded Elixir";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Live Embedded Elixir file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "leex";
    }
}

File: src/org/elixir_lang/leex/file/Type.kt
package org.elixir_lang.leex.file

import com.intellij.openapi.fileTypes.LanguageFileType

internal class Type : org.elixir_lang.eex.file.Type() {
    override fun getName(): String = "Live Embedded Elixir"
    override fun getDisplayName(): String = "LEEx"
    override fun getDescription(): String = "$name file"
    override fun getDefaultExtension(): String = "leex"

    companion object {
        val INSTANCE: LanguageFileType = Type()
    }
}
